xlb/operator/boundary_masker/mesh_boundary_masker.py

```diff
@@ -53,76 +49,70 @@ def jax_implementation(
     def _construct_warp(self):
         # Make constants for warp
         _c = self.velocity_set.c
-        _q = wp.constant(self.velocity_set.q)
+        _q = self.velocity_set.q
 
         # Construct the warp kernel
         @wp.kernel
         def kernel(
             mesh_id: wp.uint64,
-            origin: wp.vec3,
-            spacing: wp.vec3,
             id_number: wp.int32,
             bc_mask: wp.array4d(dtype=wp.uint8),
             missing_mask: wp.array4d(dtype=wp.bool),
-            start_index: wp.vec3i,
         ):
             # get index
             i, j, k = wp.tid()
 
             # Get local indices
-            index = wp.vec3i()
-            index[0] = i - start_index[0]
-            index[1] = j - start_index[1]
-            index[2] = k - start_index[2]
+            index = wp.vec3i(i, j, k)
 
             # position of the point
             ijk = wp.vec3(wp.float32(index[0]), wp.float32(index[1]), wp.float32(index[2]))
-            ijk = ijk + wp.vec3(0.5, 0.5, 0.5)  # cell center
-            pos = wp.cw_mul(ijk, spacing) + origin
-
+            pos = ijk + wp.vec3(0.5, 0.5, 0.5)  # cell center
             # Compute the maximum length
-            max_length = wp.sqrt(
-                (spacing[0] * wp.float32(bc_mask.shape[1])) ** 2.0
-                + (spacing[1] * wp.float32(bc_mask.shape[2])) ** 2.0
-                + (spacing[2] * wp.float32(bc_mask.shape[3])) ** 2.0
-            )
+            max_length = wp.sqrt(2.0) / 2.0  # half of unit cell diagonal
 
             # evaluate if point is inside mesh
-            query = wp.mesh_query_point_sign_winding_number(mesh_id, pos, max_length)
```

Comment:
please revert these changes. These are not related to this PR and we need to be more careful about this (i.e. check this more carefully). For example I am not sure what happens if we don't check for the sign. Let's have this in a separate PR not this which is related the BC profile.

Reply:
ok fine. I might have to change this again in the next PR.